rln/src/public.rs

```diff
@@ -1644,13 +1642,13 @@ mod test {
         let (identity_secret_hash_new, id_commitment_new) = keygen();
 
         // We add it to the tree
-        let identity_index_new = u64::try_from(rln.tree.leaves_set()).unwrap();
+        let identity_index_new = rln.tree.leaves_set();
         let mut buffer = Cursor::new(fr_to_bytes_le(&id_commitment_new));
         rln.set_next_leaf(&mut buffer).unwrap();
 
         // We generate a random signals
         let signal3: [u8; 32] = rng.gen();
-        let signal3_len = u64::try_from(signal3.len()).unwrap();
+        let signal3_len = signal3.len();
```

Comment:
I think this will end up with 4 bytes instead of 8 in 32 bit targets, then in line 1659 an incorrect number of bytes will be appended

Reply:
You're right, and this would lead to proofs generated by wasm fail serde in 64bit targets (I think). What we can do is either leave this functionality as-is, *or* ensure that during serde it is treated as 8 bytes(u64). What would you prefer?

Reply:
perhaps something like this could be done in line 1659 and in other places where we are appending bytes that depend on usize:
```rust
let mut signalLenVec = signal3_len.to_le_bytes().to_vec()
signalLenVec.resize(8, 0);
serialized3.append(signalLenVec);
```

Reply:
Even tho using `usize` for the code is correct, the serialization format  expects the lengths to be 8 bytes regardless of the architecture.   `someUsize.to_le_bytes()` will return an 8 bytes array in a 64bits platform, and 4 in 32 bits, so proofs generated in one platform (i.e. wasm runs 32b) wont be valid in others and viceversa

Reply:
This wont be a problem for reading since I see in this PR that it's done this way:
```rust
let id_index = usize::from_le_bytes(serialized[all_read..all_read + 8].try_into()?);
```
so variables whose value is constructed using this logic will be correct regardless of the architecture (notice that it reads 8 bytes)